src/middleware/serverMiddlewareLoader.middleware.ts

```diff
@@ -75,10 +75,11 @@ export default class ServerMiddlewareLoader {
   private prepareCSPConfig(nonce: string): ContentSecurityPolicyOptions {
     return {
       directives: {
-        defaultSrc: [`'self'`],
+        defaultSrc: [`'self'`, ServerMiddlewareLoader.extractPiwikHost(this.PIWIK_CONFIG)],
         scriptSrc: [`'self'`, 'code.jquery.com', this.CDN_HOST, `'nonce-${nonce}'`,
           ServerMiddlewareLoader.extractPiwikHost(this.PIWIK_CONFIG),
           `'sha256-+6WnXIl4mbFTCARd8N3COQmT3bJJmo32N8q8ZSQAIcU='`],
+        connectSrc: [`'self'`, ServerMiddlewareLoader.extractPiwikHost(this.PIWIK_CONFIG)],
```

Comment:
Even with the new defaultSrc value the error
`piwik.js:35 Refused to connect to 'https://matomo.platform.aws.chdev.org//piwik.php?action_name=Who%20to%20tell%20-%20Apply%20to%20strike%20off%20and%20dissolve%20a%20company%20-%20GOV.UK&idsite=13&rec=1&r=141428&h=9&m=32&s=16&url=http%3A%2F%2Fchs.local%2Fclose-a-company%2Fwho-to-tell&_id=94fac17ee6cce00d&_idn=0&_refts=0&send_image=0&pdf=1&qt=0&realp=0&wma=0&fla=0&java=0&ag=0&cookie=1&res=1440x900&pv_id=jD4lUb&pf_net=5006&pf_srv=19&pf_tfr=2&pf_dm1=70' because it violates the following Content Security Policy directive: "connect-src 'self'".`
still occurs. This seems to be necessary

Reply:
can you extract the `ServerMiddlewareLoader.extractPiwikHost(this.PIWIK_CONFIG)` result into a variable. This method gets executed multiple times